s4e-backend/src/main/java/pl/cyfronet/s4e/service/SearchService.java

```diff
@@ -43,6 +58,32 @@ public Long getCountBy(Map<String, Object> params) throws SQLException, QueryExc
         return repository.countAllByParamsMap(params);
     }
 
+    public void checkProductTypePresent(Map<String, Object> params) throws BadRequestException {
+        if (params.get(PRODUCT_TYPE) == null) {
+            throw new BadRequestException("No product type.");
+        }
+    }
+
+    public void applyLicenseByProductType(Map<String, Object> params) throws NotFoundException {
+        val product =
+                productRepository.findByName(
+                        String.valueOf(params.get(PRODUCT_TYPE)), Product.class)
+                        .orElseThrow(() -> new NotFoundException("Product not found"));
+        val accessType = product.getAccessType();
+        AppUserDetails userDetails = AppUserDetailsSupplier.get();
```

Comment:
Rather extract the AppUserDetails in controller and pass it to the method as parameter, the way it's done in most other places